core/statistics/machine/cpuStatistics.go

```diff
@@ -11,33 +11,36 @@ var durationSecond = time.Second
 
 // CpuStatistics can compute the cpu usage percent
 type CpuStatistics struct {
+	numCpu          int
 	cpuPercentUsage uint64
 }
 
+// NewCpuStatistics will create a CpuStatistics object
+func NewCpuStatistics() *CpuStatistics {
+	numCpu, _ := cpu.Counts(true)
```

Comment:
You are hiding error here. Must be careful with these things, as mentioned by Sebastian that this function might return in some situations numCpu = 0 and error. Your code will break in panic on line 41.
The constructor might return (*CpuStatistics, error) tuple.

Reply:
Done.